graveler/graveler.go

```diff
@@ -960,15 +960,19 @@ func (g *graveler) DiffUncommitted(ctx context.Context, repositoryID RepositoryI
 	if err != nil {
 		return nil, err
 	}
+	var metaRangeID MetaRangeID
 	commit, err := g.RefManager.GetCommit(ctx, repositoryID, branch.CommitID)
```

Comment:
I think in this case `branch.CommitID == ""`.  We could check that, and then be able to give a meaningful internal-ish error message if `branch.CommitID != "" && errors.Is(err, ErrCommitNotFound)`.

Reply:
Added branch's commit ID check before loading the commit information